src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,46 +1,110 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import classNames from 'classnames';
+import { getUser } from '../../api';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
 import { Loader } from '../Loader';
 
-export const TodoModal: React.FC = () => {
+type Props = {
+  todo: Todo;
+  onClose: () => void;
+};
+
+export const TodoModal: React.FC<Props> = ({ todo, onClose }) => {
+  const [user, setUser] = useState<User>();
+  const [isError, setIsError] = useState(false);
+  const [isloading, setIsLoading] = useState(true);
+
+  const {
+    id,
+    completed,
+    title,
+    userId,
+  } = todo;
+
+  useEffect(() => {
+    const getUserFromServer = async () => {
+      try {
+        const userFromServer = await getUser(userId);
+
+        setUser(userFromServer);
+      } catch {
+        setIsError(true);
+      } finally {
+        setIsLoading(false);
+      }
+    };
+
+    getUserFromServer();
+  }, []);
+
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
-
-      {true ? (
+      {isloading && (
         <Loader />
-      ) : (
+      )}
+      {((!user || isError) && !isloading) && (
```

Comment:
Hm, there is a broken layout on the error

![Screenshot 2023-04-05 at 13 16 04](https://user-images.githubusercontent.com/54940595/230055039-cd25bd90-06e7-4b7d-8823-7615519684de.png)